File: src/main/java/io/femo/http/transport/http2/frames/EndStreamFrame.java
package io.femo.http.transport.http2.frames;

import io.femo.http.transport.http2.HttpFrame;
import io.femo.http.transport.http2.HttpSettings;

/**
 * Created by felix on 9/18/16.
 */
public class EndStreamFrame extends HttpFrame {
    /**
     * Constructs a new HttpFrame with no payload and information
     *
     * @param settings the settings for the current HTTP/2,0 connection
     */
    public EndStreamFrame(HttpSettings settings) {
        super(settings);
    }

    public boolean isEndStream() {
        return (getFlags() & 1) == 1;
    }

    public void setEndStream(boolean endStream) {
        setFlags((short) (getFlags() & 0b11111110 | (endStream ? 1 : 0)));
    }
}

File: src/main/java/org/web25/http/transport/http2/frames/EndStreamFrame.kt
package org.web25.http.transport.http2.frames

import org.web25.http.transport.http2.HttpFrame
import org.web25.http.transport.http2.HttpSettings

/**
 * Created by felix on 9/18/16.
 */
open class EndStreamFrame
/**
 * Constructs a new HttpFrame with no payload and information

 * @param settings the settings for the current HTTP/2,0 connection
 */
(settings: HttpSettings) : HttpFrame(settings) {

    var isEndStream: Boolean
        get() = flags.toInt() and 1 == 1
        set(endStream) {
            flags = (flags.toInt() and 254 or if (endStream) 1 else 0).toShort()
        }
}
